twisted/conch/client/default.py

```diff
@@ -230,10 +239,10 @@ def getPrivateKey(self):
             return defer.succeed(keys.Key.fromFile(file))
         except keys.EncryptedKeyError:
             for i in range(3):
-                prompt = "Enter passphrase for key '%s': " % \
-                    self.usedFiles[-1]
+                prompt = "Enter passphrase for key '%s': " % self.usedFiles[-1]
                 try:
-                    p = self._getPassword(prompt)
+                    p = self._getPassword(prompt).encode(
+                        sys.getfilesystemencoding())
```

Comment:
_getPassword is used once where the result is passed on the wire, and once when it is used as a pass phrase when opening a ssh key. For the wire one it uses sys.getdefaultencoding() and for the file one it uses sys.getfilesystemencoding(). Since those two are different, I put the encode on the outside. But adding the encoding as a argument to _getPassword would work as well.